src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs

```diff
@@ -56,6 +56,11 @@ public interface INamedTypeSymbol : ITypeSymbol
         /// </summary>
         bool IsComImport { get; }
 
+        /// <summary>
+        /// Indicates the type is declared in source and is only visible in the file it is declared in.
+        /// </summary>
+        bool IsFile { get; }
```

Comment:
:memo: For the record, the API review for this is tracked by [https://github.com/dotnet/roslyn/issues/62254](https://github.com/dotnet/roslyn/issues/62254)